html_css/css_math_functions.md

```diff
@@ -0,0 +1,108 @@
+### Introduction
+
+You may have noticed that some property values you’ve worked with in CSS have a slightly different syntax. When the value is a word followed by a pair of parentheses () containing information between them - as in background-color: `rgb(0, 0, 0)` - you’re using CSS functions. 
+
+In this lesson, we’ll cover the basics of what a function is and some common ways they’re used in CSS.
+
+### Learning outcomes
+
+* Recognize the basic parts of a CSS function
+* Learn about the `calc()`, `min()`, `max()`, and `clamp()` functions and how to use each one
+
+##### What is a function and how are they used in CSS?
+
+As in other programming languages, functions are reusable pieces of code which perform specific tasks. Functions are passed “arguments” between parentheses, each of which is used by the function in a specific way. Some common examples are:
+
+`color: hsl(230, 100%, 50%);`
+`background: linear-gradient(90deg, blue, red);`
+
+Here, the value of `color` is the function `rgb()`, which accepts arguments in the form of numbers. It processes those numbers to calculate the rgb color corresponding to the three values given. Similarly, the `background` property has a value of `linear-gradient(90deg, blue, red)`, which calculates the angle, starting color, and ending color of a gradient.
+
+Unlike other programming languages you’ll use in TOP, CSS does not allow us to create our own functions. Instead, the language comes bundled with a list of premade functions that will help you solve the most common styling problems. [Here is a complete list](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Functions) of all CSS functions and how they are used.
```

Comment:
I'd stick this link in the 'assignment' section at the end so we don't distract our students here.

Reply:
I added an Assignments section and added the review of this link to it.